### PR TITLE
Add initial DCS APIs

### PR DESCRIPTION
**Pull request by rurounijones**
Initial implementation, submitting a PR for the simple changes so we can see if there is anywhere where we have different opinions.

Some notes:

* The file you really care about is `lib/dcs/grpc/client.rb`. You can ignore the rest.
* I am wrapping all gRPC calls in a class that hides the implementation details.
* I have renamed `out_text` to `send_message_to_all_players`. We talked about not marrying our API names and parameters to what DCS uses where we think a clearer name exists. What do you think in this case?
* I am including the generated files based off the proto defined in the current rust server. This may not be the _best_ way to do it but as we talked about in the past, good enough for now.

I am not including the Streaming events yet, want to see if there is an easy way to do it.

### CODE REVIEW

File: lib/dcs/grpc/client.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      # @see https://wiki.hoggitworld.com/view/DCS_func_outText
+      # @param text [String] the text to be displayed to the players
+      # @param display_time [Integer] the time in seconds the message will be visible
+      # @param clear_view [Bool] whether to clear existing messages when displaying this one
+      # @return [Void]
+      def send_message_to_all_players(text:, display_time: 10, clear_view: false)
+        request = Dcs::OutTextRequest.new(text: text,
+                                          display_time: display_time,
+                                          clear_view: clear_view)
+        @service.out_text(request)
```
Comment (rkusa) [inline lib/dcs/grpc/client.rb:40]: For consistency between clients, should we also rename the server/service method here?

File: lib/dcs/grpc/client.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      #
+      # Displays the specified text to all players on the DCS server in the top right
+      # of the game window. This maps to the `outText` command in DCS.
+      #
+      # @see https://wiki.hoggitworld.com/view/DCS_func_outText
+      # @param text [String] the text to be displayed to the players
+      # @param display_time [Integer] the time in seconds the message will be visible
+      # @param clear_view [Bool] whether to clear existing messages when displaying this one
+      # @return [Void]
+      def send_message_to_all_players(text:, display_time: 10, clear_view: false)
```
Comment (rkusa) [inline lib/dcs/grpc/client.rb:36]: I am fine with renaming it, the only nit I have is that the name feels a bit too long, how about shortening it to one of the following: `message_to_all`, `send_message_to_all`, `notify_all`?

**Review by rkusa — APPROVED** [review]
👍

File: lib/dcs/grpc/client.rb (hunk 1 above)
Comment (rurounijones) [reply lib/dcs/grpc/client.rb:40]: I am honestly not sure, I can see arguments for having the server-side API use the same names as the Lua APIs and also arguments for changing them to match the clients.

I am fine with either option.

File: lib/dcs/grpc/client.rb (hunk 2 above)
Comment (rurounijones) [reply lib/dcs/grpc/client.rb:36]: There are coaliation and group message APIs so I `message_to_all`, `message_to_coalition`, `message_to_group` sounds like a plan. I will change it.

File: lib/dcs/grpc/client.rb (hunk 1 above)
Comment (rkusa) [reply lib/dcs/grpc/client.rb:40]: True, there are arguments for either option. So we can probably just keep it as is for now

**Pull request merged**